src/core/services.py

```diff
@@ -508,3 +511,73 @@ def queue_job(project_id: int, job_id: int, session: Session) -> None:
         logger.error(
             f"Cannot queue job {job_id}, it's of status {job.status()}."
         )
+
+
+def get_directory_listing(path: str) -> Dict[str, Any]:
+    """Get contents found in a given directory. Does not search recursivly.
+
+    Parameters
+    ----------
+    path    :   str
+        Path to the folder one wants to get listing from.
+
+    Raises
+    ------
+    NotADirectoryError
+        When the given path is a file and not a directory.
+
+    FileNotFoundError
+        Directory cannot be found at a given path.
+
+    Returns
+    -------
+    Dict
+        jsTree formatted json containing information about root node at path.
+    """
+    root_node: Dict[str, Any] = dict()
+    child_list = []
+
+    if isfile(path):
+        raise NotADirectoryError("Path must be a directory.")
+
+    if not isdir(path):
+        raise FileNotFoundError("Directory at '{}' was not found.".format(path))
+
+    root, dirs, files = next(os.walk(path))
+
+    # Ensure alphabetical order
+    dirs.sort()
+    files.sort()
+
+    # Append directories
+    child_list.extend(
+        [
+            {
+                "text": name,
```

Comment:
Wouldn't this make a little more sense?

```suggestion
                "name": name,
```

Reply:
Ref.: https://www.jstree.com/docs/json/